src/basic/mount-util.c

```diff
@@ -874,3 +878,77 @@ int mount_propagation_flags_from_string(const char *name, unsigned long *ret) {
                 return -EINVAL;
         return 0;
 }
+
+int mount_option_mangle(const char *options, unsigned long *ret_mount_flags, char **ret_remaining_options) {
+        const struct libmnt_optmap *map;
+        _cleanup_free_ char *ret = NULL;
+        unsigned long mount_flags;
+        const char *p;
+        int r;
+
+        /* This extracts mount flags from the mount options, and store
+         * non-mount-flag options to *ret_remaining_options.
+         * E.g.,
+         * "rw,nosuid,nodev,relatime,size=1630748k,mode=700,uid=1000,gid=1000"
+         * is split to MS_NOSUID|MS_NODEV|MS_RELATIME and
+         * "size=1630748k,mode=700,uid=1000,gid=1000".
+         * See more examples in test-mount-utils.c.
+         *
+         * Note that *ret_mount_flags is not initialized in this function.
+         * The caller needs to initialized to zero or some default vaule. */
+
+        assert(options);
+        assert(ret_mount_flags);
+        assert(ret_remaining_options);
+
+        mount_flags = *ret_mount_flags;
```

Comment:
hmm, not a fan of parameters that are used for input and output like this... I think I'd prefer it if we'd pass this in in a separate input parameter, i.e.:

```c
int mount_option_mangle(const char *options, unsigned long mount_flags, char **ret_remaining_options, unsigned long *ret_mount_flags);
```